### PR TITLE
Test compatibility with Python generated metadata

### PR DESCRIPTION
**Pull request by lmars**
This is the first half of #26.

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -58,14 +70,41 @@ func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) error {
...
+	m := data.FileMeta{Length: n, Hashes: make(map[string]data.HexBytes, len(hashes))}
+	for hashFunc, h := range hashes {
+		m.Hashes[hashFunc] = h.Sum(nil)
+	}
+	return m, nil
+}
+
+func NormalizeTarget(path string) string {
+	s := filepath.Clean(path)
+	if s[0] == '/' {
```
Comment (titanous) [inline util/util.go]: It can't hurt to make sure the length is greater than zero before doing this.

File: data/types.go [hunk 2]
```diff
@@ -104,6 +104,14 @@ type FileMeta struct {
 	Custom map[string]interface{} `json:"custom,omitempty"`
 }
 
+func (f FileMeta) HashFuncs() []string {
```
Comment (titanous) [inline data/types.go]: I'd prefer calling this `HashAlgorithms`

File: client/client.go [hunk 3]
```diff
@@ -447,14 +447,14 @@ func (c *Client) Download(name string, dest Destination) (err error) {
...
-	localMeta, ok := c.targets[name]
+	// return ErrUnknownTarget if the file is not in the local targets.json
+	localMeta, ok := c.targets[util.NormalizeTarget(name)]
 	if !ok {
 		return ErrUnknownTarget{name}
 	}
 
 	// get the data from remote storage
-	r, size, err := c.remote.Get("targets/" + name)
+	r, size, err := c.remote.Get("targets" + name)
```
Comment (titanous) [inline client/client.go]: Use `path.Join`?

File: client/client_test.go [hunk 4]
```diff
@@ -713,7 +722,7 @@ func (s *ClientSuite) TestDownloadTargetCorruptData(c *C) {
 	remoteFile := s.remote["targets/foo.txt"]
 	remoteFile.buf = bytes.NewReader([]byte("corrupt"))
 	var dest testDestination
-	assertWrongHash(c, client.Download("foo.txt", &dest))
+	assertWrongHash(c, client.Download("/foo.txt", &dest))
```
Comment (titanous) [inline client/client_test.go]: Perhaps `Download` should prepend the slash if it doesn't exist?

**Comment by lmars** [comment]
@titanous comments addressed.

`Download` does normalize the given file name, but given the manifest uses "absolute" paths, I think the client should do the same when requesting files.

**Comment by titanous** [comment]
LGTM

**Pull request merged**